src/snyk/base/services/scanModeService.ts

```diff
@@ -0,0 +1,55 @@
+import { EXECUTION_PAUSE_INTERVAL, EXECUTION_THROTTLING_INTERVAL } from '../../common/constants/general';
+import { SNYK_CONTEXT, SNYK_MODE_CODES } from '../../common/constants/views';
+import { IContextService } from '../../common/services/contextService';
+
+export class ScanModeService {
+  private _mode = SNYK_MODE_CODES.AUTO;
+  // Platform-independant type definition.
+  private _unpauseTimeout: ReturnType<typeof setTimeout> | undefined;
```

Comment:
style comment: it's usually recommended to initialise empty values to `null` rather than `undefined`, since `null` is set by program logic rather than `undefined` which can also mean non-existing variable. For your consideration :)

Reply:
This code doesn't do initialisation, rather defining that the property is optional. Another way to put it would be by placing a question mark at the property declaration. TypeScript uses `undefined` as a default value for non-initialised class members as per [their guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined). 

I see your point and it makes sense, however this code is not new, so I'd prefer not to do any changes around it and keep it until it needs to be changed.